controlplane/pkg/nsmd/nsmd.go

```diff
@@ -574,3 +580,29 @@ func (nsm *nsmServer) StartAPIServerAt(ctx context.Context, sock net.Listener, p
 	}()
 	span.Logger().Infof("NSM gRPC API Server: %s is operational", sock.Addr().String())
 }
+
+// WorkspaceFromContext finds workspace by clientToken stored in the Context
+func (nsm *nsmServer) WorkspaceFromContext(ctx context.Context) *Workspace {
+	if clientToken := tools.ClientToken(ctx); len(clientToken) > 0 {
+		for _, w := range nsm.workspaces {
```

Comment:
`workspaces` is a map: Name -> Workspace, how often do we get Workspace by its name? Maybe makes sense to create map: Token -> Workspace?

Reply:
I found 2 places where a workspace is looked up by its name. It doesn't really seem convincing so far to change the key to token.